presto-hive/src/main/java/com/facebook/presto/hive/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -157,26 +159,29 @@ else if (type.equals(BIGINT) && (statistics instanceof LongStatistics || statist
             }
             return createDomain(type, hasNullValue, parquetIntegerStatistics);
         }
-        else if (type.equals(DOUBLE) && (statistics instanceof DoubleStatistics || statistics instanceof FloatStatistics)) {
-            ParquetDoubleStatistics parquetDoubleStatistics;
-            if (statistics instanceof DoubleStatistics) {
-                DoubleStatistics doubleStatistics = (DoubleStatistics) statistics;
-                // ignore corrupted statistics
-                if (doubleStatistics.genericGetMin() > doubleStatistics.genericGetMax()) {
-                    return Domain.create(ValueSet.all(type), hasNullValue);
-                }
-                parquetDoubleStatistics = new ParquetDoubleStatistics(doubleStatistics.genericGetMin(), doubleStatistics.genericGetMax());
-            }
-            else {
-                FloatStatistics floatStatistics = (FloatStatistics) statistics;
-                // ignore corrupted statistics
-                if (floatStatistics.genericGetMin() > floatStatistics.genericGetMax()) {
-                    return Domain.create(ValueSet.all(type), hasNullValue);
-                }
-                parquetDoubleStatistics = new ParquetDoubleStatistics((double) floatStatistics.getMin(), (double) floatStatistics.getMax());
+        else if (type.equals(DOUBLE) && statistics instanceof DoubleStatistics) {
+            DoubleStatistics doubleStatistics = (DoubleStatistics) statistics;
+            // ignore corrupted statistics
+            if (doubleStatistics.genericGetMin() > doubleStatistics.genericGetMax()) {
+                return Domain.create(ValueSet.all(type), hasNullValue);
             }
+            ParquetDoubleStatistics parquetDoubleStatistics = new ParquetDoubleStatistics(doubleStatistics.genericGetMin(), doubleStatistics.genericGetMax());
             return createDomain(type, hasNullValue, parquetDoubleStatistics);
         }
+        else if (type.equals(FLOAT) && statistics instanceof FloatStatistics) {
```

Comment:
None of the other types do that, so I'm hesitant to add it. And if you have the wrong type of statistics then you have corrupted statistics of some sort, so it's reasonable to not do pushdown.